### ISSUE TITLE
Syncing onto https://www.samba.org/ftp/ccache/

### ISSUE THREAD

**Issue by rubyFeedback** [ccache/ccache#396]
Hello ccache devs,

I have been using this URL:

https://www.samba.org/ftp/ccache/?C=M;O=D

respectively:

https://www.samba.org/ftp/ccache/ccache-3.6.tar.xz

For obtaining the latest ccache. I update program versions
from the commandline, e. g. a script called "update_entry"
so I just pass the name of the program (ccache) into it.
So far so fine. I noticed that there is a new release of
ccache out and ran the script but it could not find a new
release (because the script is quite stupid and only checks
for the main FTP site that I registered in it - so in this
example, the main ftp site of samba, for ccache).

On GitHub here, though, you can download the latest
source at:

https://github.com/ccache/ccache/releases/download/v3.7/ccache-3.7.tar.xz

I downloaded this, compiled it, installed fine, and compiled several
other programs via ccache (I use ccache since quite a long time
since it appears to really help when you have to compile a lot of
things every week, as I do).

Now to my specific suggestion:

- Would it be possible to automatically sync any release on github,
back to the samba ftp site? This may be trivial via some script and/or
hooks.

Note that while I have a slight preference to preferentially retain 
URLs that are not github (for several reasons), including the 
samba-ftp cache site, I don't really mind either way, whether the ftp
site is at the old place, or at the new place, or both. But in particular
when there is a delay, or a lack of syncing/upload to one site, this
causes a slight confusion in my scripts right now. Now, this can be
resovled in my own code, no problem; I can just add code that
will also check on github in the long run - but I still did want to mention
it because it seems as if there will be a casual push towards using
only github and eventually fading out the old ways, which I am not
sure is necessarily a great thing to do (but I can also understand when
there is more work involved, which is why scripts should ideally be
used to automatically do that; I assume that other projects may
face similar problems, so I am grateful that ccache provides a
rich alternative of downloads e. g. at https://github.com/ccache/ccache/releases -
other projects often only have two downloads which we users then
still have to re-name, since the extract tarball does not match with
the name of the program at hand).

This is really a tiny issue but I think it is still better to at the least report
it so that you at upstream can consider, as it is also somewhat related
to the whole samba suite how they want to make their software available.
Again - I have no real particular strong preference, but I believe it is better
to make it available on all sites at the same time, if possible, rather than 
keep other sites secondary (or eventually fade it out anyway even if noone
actively wanted to do so - we are all creatures of habit).

**Comment by jrosdahl** [comment]
Hello!

I don't know if you're on the ccache mailing list, but anyway: From
https://www.mail-archive.com/ccache@lists.samba.org/msg01463.html: _"Note: New release files will no longer be uploaded to https://www.samba.org/ftp/ccache, but old releases will remain there."_

So it's intentional to not upload releases to https://www.samba.org/ftp/ccache anymore.

> Would it be possible to automatically sync any release on github, back to the samba ftp site? This may be trivial via some script and/or hooks.

Yes, that would be possible, but I have stopped doing that unless there is some significant disadvantage of using GitHub releases instead.

> it is also somewhat related to the whole samba suite how they want to make their software available.

Well, this is actually related to why I have moved both the ccache web site and where the releases are located from samba.org... ccache has not been part of the Samba suite since I took over as the maintainer in 2010; I'm not part of the Samba team.
